scripts/validate_data.py

```diff
@@ -1,71 +1,71 @@
-""" Python script to validate data
-
-Run as:
-
-    python3 scripts/validate_data.py data
-"""
-
 from pathlib import Path
 import sys
 import hashlib
 
 
 def file_hash(filename):
-    """ Get byte contents of file `filename`, return SHA1 hash
-
+    """ Get SHA1 hash of the contents of a file
     Parameters
     ----------
     filename : str
         Name of file to read
-
     Returns
     -------
     hash : str
         SHA1 hexadecimal hash string for contents of `filename`.
     """
-    # Open the file, read contents as bytes.
-    # Calculate, return SHA1 has on the bytes from the file.
-    # This is a placeholder, replace it to write your solution.
-    raise NotImplementedError(
-        'This is just a template -- you are expected to code this.')
+    sha1 = hashlib.sha1()
+    with open(filename, 'rb') as fobj:
+        data = fobj.read()  # Read the entire file at once
+        sha1.update(data)
+    return sha1.hexdigest()
 
 
 def validate_data(data_directory):
     """ Read ``data_hashes.txt`` file in `data_directory`, check hashes
-
     Parameters
     ----------
     data_directory : str
         Directory containing data and ``data_hashes.txt`` file.
-
```

Comment:
The tradition for these docstrings is to separate the sections with  new lines - I have never tried deleting them, but in any caseI think they are easier to read with:

https://numpydoc.readthedocs.io/en/latest/format.html